### DISCUSSION TITLE
SMT PCB design

### DISCUSSION THREAD

**Discussion by eMadman** [doudar/SmartSpin2k#326] [Ideas]
The through-hole PCB is great, there's a lot of components to solder in place by hand and the wiring harness can be a pain to crimp and solder.

This discussion thread is here to explore an SMT design for the PCB.

Some requirements:
- 3.5mm (shifter) and DC power connector should be PCB mounted - **Using a breakout cable with 8 pin mini din**
- Power and shifter connectors should ideally face towards the front of the bike or down towards the bottom **Breakout cable solves this**
- We may need to make case design/length adjustments to allow the power and shifter connectors to exit without interfering with the knob if they face downwards.  **Breakout cable solves this**
- Parts should be easily sourced through pcbway - ideally, we want to be able to get to 1-click ordering.

MCU: ESP32 - we need the bluetooth and wifi modules
Stepper Motor controller: TMC 2209 or 2225.  From what I recall, the 2208 tends to run hot in stepstick form factor (small copper pours on several stepsticks).  Maybe the 2130?  I'll let @doudar confirm options.  The biggest draw for the TMC's is the benefit of stealthchop and UART.  2130 brings sensorless stall detection but requires code to support it  **2109**

**Comment by doudar** [comment]
It looks like the max dimension is ~58mm X ~60mm depending on the orientation of the board and where the power and shifter connectors line up.

Additional things to think about:
- I think 2209 is best suited due to features and known compatibility with the current code.
- Should the design use a bare esp32 ic vs the esp32 module that the devkit uses? **Probably a module**
- Is adding psram a viable (or necessary) option? **no**
- Should there be a serial input added as well for future Peloton power decoding? **yes, if space available**
- Adding a hardware de-bounce circuit (opto-isolator) to the PCB would be very beneficial.
- Should we explore adding a NRF52 to add ANT+ so that we could use a heavier (better) web server?
- Should all of the (existing) connectors be located at the end of one breakout cable? **yes**
- cp2102 serial driver

![image](https://user-images.githubusercontent.com/17362216/153880618-2ac920db-6ae3-4a0f-9963-b313e609ef28.png)


![image](https://user-images.githubusercontent.com/17362216/153880202-cc74eafa-5a70-4fa6-b5d5-56b2bac5ea5a.png)

**Comment by doudar** [comment]
Adding that we really need to evaluate the connections required for Peloton input (if we decide to implement it) as well as complexity and availability of a breakout cable if we decide to use that as well. 

We wouldn't want to create a new problem by implementing a PCB that now requires a special hard to obtain or manufacture breakout cable. 

@eMadman , have you looked into breakout cables at all yet?

**Reply by eMadman** [reply]
Assumptions:

- DC In- 2-pin 24awg high strand count or 22awg of the cheaper stuff.  
- Shifter: 3 pin - left/right/gnd
- Peloton serial input: 3-pin
- Peloton serial output: 3-pin

Considerations: 

- For the Peloton serial and shifter, we might be able to share GND since I think those can all run off 5v
- Can we use a stereo to stereo splitter for the Peloton?  Might save some conductors
- On the PCB side, we need between 9 conductors (assuming a separate plug for peloton in/out and shared gnd... more if we need to separate those out) on some sort of header XH, Microfit 3, etc..  XH connectors are most easily sourced.  Microfits tend to be easier to get originals.  Plenty of other options, but it's good to use something ubiquitous in hobby or PC building 
- Coming out of the chassis... it seems like the best bet will be a strain relieved cable terminating in separate plugs - DC, Serial in, serial out, shifter.
- with multiple 3.5mm connectors, it will be a challenge to know which one is which.  Can't correct this in software since Peloton serial needs to go through the level converter.  Also makes it more of a pain to quickly mount/dismount.
- Possible opportunity for a wiring harness to combine all the 3.5 and DC sources into a single connector that meets the wiring harness exiting the case - makes for easier subsequent teardowns but adds cost/complexity and hurts portability (something like a 12 conductor lemo)

Connector ideas:
Mini Din connectors have the pins necessary for what we need - we think the peloton serial is a split connection so we don't need the pins for in/out direct from the PCB

Single connector out of the box is possible:
https://www.mouser.ca/ProductDetail/Kycon/KMDGX-8S-N?qs=SY6bF4iPirQpujXD5r7QXg%3D%3D

Cable coming out of the box is attractive from a repairability standpoint:
https://www.alibaba.com/product-detail/CUSTOM-1M-PC-Engine-Controller-S_1600188715945.html

Peloton could use a 2.5 > 3.5mm y splitter instead
https://www.amazon.com/Homyl-Splitter-Extension-Earphone-Headset/dp/B07FNS3SRX

![image](https://user-images.githubusercontent.com/24726844/155613810-ab0c8d8f-3fae-4902-ad93-85f5a6acf392.png)

**Reply by h2zero** [reply]
> Cable coming out of the box is attractive from a repairability standpoint:
https://www.alibaba.com/product-detail/CUSTOM-1M-PC-Engine-Controller-S_1600188715945.html

Love this idea, and thanks for the link 😀, could be useful in my future endeavors.

**Comment by doudar** [comment]
![ss2k Cable brainstorm 2](https://user-images.githubusercontent.com/17362216/155751578-d83d58f3-bb05-4e9a-bc8a-0d8c3ed37758.png)

Rough draft of the current components required on the pcb.

**Reply by doudar** [reply]
@alexquilty do you have anything to add or questions on the above?

**Reply by doudar** [reply]
@fireyeti or @MarkusSchneider , or @h2zero, anything to add?

**Reply by doudar** [reply]
Actually it just hit me that @h2zero may have some insight into custom wire harnesses and the applicability and possibility of getting them made.

**Reply by h2zero** [reply]
I would stick to COTS on this, custom would be too much $$$$ in my opinion, unless you're making 10's of thousands. But I could be wrong.

**Reply by h2zero** [reply]
Do you plan to use ant+ here as well? There's a company 30 mins from me that specializes in this.

**Reply by doudar** [reply]
> Do you plan to use ant+ here as well? There's a company 30 mins from me that specializes in this.

We might. We don't know if there will be room on the PCB for the NRF52 as well as how to do update the NRF52 via serial from the ESP32. Might be too risky on the risk/reward scale. The benefit is we could possibly offload the BLE as well which solves all those WiFi coexistence problems.

**Reply by fireyeti** [reply]
> @fireyeti or @MarkusSchneider , or @h2zero, anything to add?

Is there any advantage to USB-C? Micro USB seems very usable with great support and everyone has cables. I think micro is likely a smaller footprint than C. Regarding the custom cable, is there a good vendor to source from? @eMad might have insight here?

**Reply by doudar** [reply]
Here's the link to a quick 1 day hack of my last attempt of the PCB. It may or may not work as designed. The form factor has probably changed and I know some of the design requirements have changed, but feel free to take a look at it and glean any info from it you can. 

https://easyeda.com/editor#id=|f1861afb4e6f4e6789b6c8446c331680|315cd9a1f25b4591b23bc13ca9c15b58
![PCB_PCB_New Project_2022-02-25 (1)](https://user-images.githubusercontent.com/17362216/155825357-d378b198-0c54-49e1-9cf9-862edcacd209.png)

![Schematic_SmartSpin2K_2022-02-25](https://user-images.githubusercontent.com/17362216/155825370-faf62ee7-9975-4dbf-baee-2f3a4175599d.png)
[BOM_SmartSpin2K_2022-02-25.csv](https://github.com/doudar/SmartSpin2k/files/8145712/BOM_SmartSpin2K_2022-02-25.csv)

**Reply by h2zero** [reply]
> > Do you plan to use ant+ here as well? There's a company 30 mins from me that specializes in this.
> 
> We might. We don't know if there will be room on the PCB for the NRF52 as well as how to do update the NRF52 via serial from the ESP32. Might be too risky on the risk/reward scale. The benefit is we could possibly offload the BLE as well which solves all those WiFi coexistence problems.

Definitely more expensive, a second esp32 is cheaper if BLE only, use the c3 if going this route.

**Reply by doudar** [reply]
> > > Do you plan to use ant+ here as well? There's a company 30 mins from me that specializes in this.
> > 
> > 
> > We might. We don't know if there will be room on the PCB for the NRF52 as well as how to do update the NRF52 via serial from the ESP32. Might be too risky on the risk/reward scale. The benefit is we could possibly offload the BLE as well which solves all those WiFi coexistence problems.
> 
> Definitely more expensive, a second esp32 is cheaper if BLE only, use the c3 if going this route.

That's an interesting idea...

**Reply by doudar** [reply]
> > > Do you plan to use ant+ here as well? There's a company 30 mins from me that specializes in this.
> > 
> > 
> > We might. We don't know if there will be room on the PCB for the NRF52 as well as how to do update the NRF52 via serial from the ESP32. Might be too risky on the risk/reward scale. The benefit is we could possibly offload the BLE as well which solves all those WiFi coexistence problems.
> 
> Definitely more expensive, a second esp32 is cheaper if BLE only, use the c3 if going this route.

Do you have any input (preference) on using an ESP32 module vs the bare chip in the design? If using the bare chip, will that decrease the PCB size, or do we risk running into antenna design issues? (or other issues?)

**Reply by h2zero** [reply]
Module only, never bare chip (unless you have an RF engineer and want to deal with all that...). There's a long list of reasons that I won't bother listing (think FCC testing etc.), KISS for this absolutely.

**Reply by doudar** [reply]
> Module only, never bare chip (unless you have an RF engineer and want to deal with all that...). There's a long list of reasons that I won't bother listing (think FCC testing etc.), KISS for this absolutely.

That's great input! 

Would you be willing to interact with our senior EE guys at Purdue to help guide them on their design project? I know you're busy, but you have the most electronics experience (especially esp32) of anyone involved in this project (the regulars). It would probably only involve the occasional email question and maybe a Zoom call before they finish their first circuit design.

**Reply by h2zero** [reply]
Haha sure, you can give them my email info. Looks like you've got good progress on this with your base design, not sure how much input I can add. As long as the ground planes stay away from the antenna (probably distance the wiring from it also) all is well 😄.

**Reply by doudar** [reply]
Power supply info:
- Power requirements 500ma-1000ma
- Make sure to use all ceramic capacitors.
- max 17504 as part of the power supply - cost providing....
- another cheaper power supply option: MFR.Part #TPS5430DDAR JLCPCB Part #C9864
- Inductor Suggestion for the above MFR.Part #CKCS3012-15uH/MCopy JLCPCB Part #C354559
- Another Power Supply option: MFR.Part # TPS54331DR JLCPCB Part #C9865
